### PR TITLE
Fully Developed Application

### PR DESCRIPTION
**Pull request by El-Potato-Slayer**
- Contains input that user can enter a city name
- Notifies user if entered input is incorrect
- Most important weather information is displayed

### CODE REVIEW

**Review by krys2fa — CHANGES_REQUESTED** [review]
## Awesome work @El-Potato-Slayer 👏 
I liked the images used to change appearance based on the data received. 💪 
Only one requirement was missed.
### Status: Changes Required ♻️ 

- [x] [REQUIRED] Even though you are displaying temperatures in both Celsius and Fahrenheit, you should add a toggle control to determine if the result will be shown in Celsius or Fahrenheit.
**Hint**: you can make two different requests for the Celsius data and the Fahrenheit data OR make only one request and use math to convert it to the other unit ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))

Please request another review after implementing the change.

Happy Coding 🖥️  😃 

###### You can contact me on slack (Chris Amanor) for any clarification.

File: dist/styles.css [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  left: 50%;
+  transform: translate(-50%, -50%);
+  transition: all 0.5s ease-in-out;
+}
+
+#weather-info{
+  top: 50%;
+  left: -10%;
+  width: max-content;
+  /* right: 50%; */
```
Comment (rukundoeric) [inline dist/styles.css:18]: remove this comment

File: dist/styles.css [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+.switch input{
+  opacity: 0;
+  height: 0;
+  width: 0;
+}
+
+/* .slider input:checked + .slider{
+  background: aqua;
+} */
```
Comment (rukundoeric) [inline dist/styles.css:52]: remove this comment

File: src/frontend/helpers.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+export default function removeChildren(parent) {
+  while (parent.firstChild) {
+    parent.firstChild.remove();
+  }
+}
+
+export function switchTemperature() {
+  const button = document.querySelector('.switch');
+  const chk = document.querySelector('.switch input');
+  // chk.checked = false
```
Comment (rukundoeric) [inline src/frontend/helpers.js]: remove this comment

**Review by rukundoeric — CHANGES_REQUESTED** [review]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  - [x] The project is clean. 💯
  - [x] The project follows Git flow. 💯

- ### To Improve

  - [ ] [MANDATORY] When I click on Toggle control after getting the results, It should still be able to toggle temperatures accordingly. What I mean is, I don't need to click on seach again just to toggle between  Celsius or Fahrenheit. 
  - [ ] [MANDATORY]  Remove all commented codes.

_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

**Review by rukundoeric — CHANGES_REQUESTED** [review]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  - [x] The project is clean. 💯
  - [x] The project follows Git flow. 💯

- ### To Improve

  - [ ] [MANDATORY] Toggle control is still not working properly. refer to this point from the previous review ```When I click on Toggle control after getting the results, It should still be able to toggle temperatures accordingly. What I mean is, I don't need to click on seach again just to toggle between  Celsius and Fahrenheit``` 

_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

**Review by Srjnnnn — CHANGES_REQUESTED** [review]
## Status: Changes Required :recycle: 

Hello @El-Potato-Slayer. Your page looks good definitely. I think you've understood the APIs and the general React flow. :muscle:. However, before moving on you need to solve the toggle button issue.

- [ ]  Please refer to the last two comments on this PR and arrange a toggle functionality without going back to the main page or refresh. It is a core functionality about React that you should grasp. That's why we insist you do this.

Apart from this, I believe you are on a good path. Please fix this issue and submit for a re-review.
(PS: If you feel stuck and can't figure out how to do this, you always can arrange a one-on-one with a Full-Time-TSE.)

If you have any questions about the review, feel free to Slack me. :100: 
_Eyüp Sercan UYGUR_

**Review by MuriloRoque — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello @El-Potato-Slayer :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Pull request merged**